src/scripts/targetingex.ts

```diff
@@ -105,10 +103,10 @@ namespace Scripts {
          * @param distance max vzdalenost, vyuzivaji jen nektere aliasy a je pretizeno pokud je zadano primo u aliasu. vychozi 20
          * @returns vraci Traget result, pokud neni nalezeno je prazdny tj. success= false
          */
-        static getTarget(targets: string, distance?: number): TargetResult | undefined {
+        static getTarget(targets: string | TargetEnum, distance?: number): TargetResult | undefined {
```

Comment:
tady by stalo za zminku to trochu zprehlednit, pouzit f tech podminkach check ` === TargetEnum.self` atd..

Reply:
Zvazoval sem to, nicmene nektere vyhodnocuji jako string a pracuji i hodnotami, ktere nesou enumem definovatelne. narp. 
"nearinjuredalie,5", kde 5 je maxdistance. Ale kouknu nato.

Reply:
nešlo by tie parametre posielať ako objekt? namiesto stringu? tak to funguje aj u ostatných viac parametrizovateľných veciach

Reply:
Genialni ;], vcera sem to prave predelal, uz puvodne jsem ty chtel udelat jako objekt, vytvoril sem na to Interface ITargetAlias, idealni by byli implicitni konverze ze stringu i z enumu, ale ty sem jeste nezkoumal v TS.

Reply:
Pro predstavu
```        static getTarget(targets: string|TargetEnum|Array<ITargetAlias>, maxDistance?: number): TargetResult | undefined {
            TargetingEx.cancelResetTarget();
            let result: TargetResult = new Scripts.TargetResult();
            const targetAliases = new Array<ITargetAlias>();

            if (targets && typeof targets === 'string') {
                for (const target of  targets.split('|')) {
                    targetAliases.push(TargetingEx.parseTargetAlias(target));
                }
            } else if (targets && targets.length) {
                targetAliases.push(...<Array<ITargetAlias>>targets);
            } else if (targets) {
                targetAliases.push(<ITargetAlias>{ alias: targets })
            }
               
            for (const target of targetAliases) {
                const maxDisntace = target.maxDistance ?? maxDistance ?? 20;
                result = new Scripts.TargetResult();

                if (target.alias === TargetEnum.self) {```

Reply:
```
cast("Magic Arrow", TargetEnum.self);
cast("Magic Arrow", "hover|self");
cast("Magic Arrow", [ { alias: 'hover', maxDistance: 5 }, { alias: 'self' } ]);
```